src/main/java/games/strategy/triplea/ui/menubar/LobbyMenu.java

```diff
@@ -298,38 +303,38 @@ private long requestTimespanSupplication() {
             JOptionPane.OK_CANCEL_OPTION, JOptionPane.QUESTION_MESSAGE, null, timeUnits.toArray(),
             timeUnits.toArray()[3]);
     if (result < 0) {
-      return -1;
+      return Duration.ofMillis(-1);
     }
     final String selectedTimeUnit = (String) timeUnits.toArray()[result];
     if (selectedTimeUnit.equals("Forever")) {
-      return Long.MAX_VALUE;
+      return null;
```

Comment:
would be excellent to avoid returning null, helps avoid NPE's and null handling. Since this is a private method, we may be able to move the return null logic to the caller. Failing that, a reasonable last resort option would be to return an `Optional<TemporalAmount>` instead of `TemporalAmount`.  In part this then helps the return value from having multiple meanings, empty meaning one specific condition was hit otherwise it is value object with a very different semantic meaning. The semantic meaning of a variable changing based only on its value adds a decent level of complexity, makes it harder to reason about the logic.